api/nodes/serializers.py

```diff
@@ -1094,6 +1095,11 @@ def validate_data(self, node, user_id=None, full_name=None, email=None, index=No
             raise exceptions.ValidationError(detail='A user ID or full name must be provided to add a contributor.')
         if user_id and email:
             raise exceptions.ValidationError(detail='Do not provide an email when providing this user_id.')
+        if email:
+            try:
+                validate_email(email)
```

Comment:
There's different places we can do this validation! `NodeContributorsCreateSerializer.validate_data`, as you chose here, is one option, but I'm thinking we should pull this back onto either 1) a validator on the User.username field or 2) the `ContributorMixin.add_unregistered_contributor` method.  Since there are other places where we're calling `add_unregistered_contributor` outside of this particular endpoint, we can more broadly fix this issue in one location.

There might be a reason we don't have this validator on the field already (#1) - and to implement, we'd have to add a schema migration and potentially clean up bad emails (unsure about this part). So I'm leaning towards #2 - adding the validation within `add_unregistered_contributor`.  We don't need this to get too out of scope.

You would move to catching this BlacklistedEmailError in `NodeContributorsCreateSerializer.create`.